test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/TestAutoCreateSharedArchiveUpgrade.java

```diff
@@ -141,4 +164,94 @@ static void assertJSAVersionMismatch(OutputAnalyzer output) {
+                build = 29;
+                break;
+            default:
+                build = 0;
+                break;
+        }
+        // Get correct file name for architecture
+        switch(arch) {
+            case("x86"):
+            case("x86_64"):
+            case("amd64"):
+                architecture = "x";
+                break;
+            case("aarch64"):
+                architecture = "aarch";
+                break;
+            default:
+                architecture = "";
+                break;
+        }
+        // File name is bundles/<os>-<architecture>64/jdk-<version>_<os>-<architecture>64_bin.<extension>
+        // Ex: bundles/linux-x64/jdk-19_linux-x64_bin.tar.gz
+        switch (osID) {
+            case "Windows":
+                jdkArtifactMap.put("version", version);
+                jdkArtifactMap.put("build_number", build);
+                jdkArtifactMap.put("file", "bundles/windows-x64/jdk-" + version + "_windows-x64_bin.zip");
+                return fetchBootJDK(jdkArtifactMap, version);
+
+            case "MacOSX":
+                jdkArtifactMap.put("version", version);
+                jdkArtifactMap.put("build_number", build);
+                jdkArtifactMap.put("file", "bundles/macos-" + architecture + "64/jdk-" + version + "_macos-" + architecture + "64_bin.tar.gz");
+                return fetchBootJDK(jdkArtifactMap, version);
+            case "Linux":
+                jdkArtifactMap.put("version", version);
+                jdkArtifactMap.put("build_number", build);
+                jdkArtifactMap.put("file", "bundles/linux-" + architecture + "64/jdk-" + version + "_linux-" + architecture + "64_bin.tar.gz");
+                return fetchBootJDK(jdkArtifactMap, version);
+
+            default:
+                return DEFAULT_BOOT_JDK;
+        }
+    }
+
+    // Fetch JDK version from artifactory
+    private static String fetchBootJDK(HashMap<String, Object> jdkArtifactMap, int version) {
+        String path = DEFAULT_BOOT_JDK;
+        try {
+            path = ArtifactResolver.resolve("jdk", jdkArtifactMap, true) + "/jdk-" + version;
+            System.out.println("Boot JDK path: " + path);
+        } catch (ArtifactResolverException e) {
+            Throwable cause = e.getCause();
+            if (cause == null) {
+                System.out.println("Cannot resolve artifact, "
+                        + "please check if JIB jar is present in classpath.");
+            } else {
+                throw new RuntimeException("Fetch artifact failed: "
+                        + "\nPlease make sure the artifact is available.", e);
+            }
+        }
+        return path;
+    }
+
+    private static String getOsId() {
+        String osName = props.getProperty("os.name");
+        if (osName.startsWith("Win")) {
+            osName = "Windows";
+        } else if (osName.equals("Mac OS X")) {
+            osName = "MacOSX";
+        }
+        return osName;
+    }
```

Comment:
Instead of writing your own code for detecting OS/architecture, etc, you should use [test/lib/jdk/test/lib/Platform.java
](https://github.com/openjdk/jdk/blob/master/test/lib/jdk/test/lib/Platform.java)